T0096 gift nordic bug

simgonzalez opened the pull request:
Correct the gift send without settings for payments.

Commit:
T0096 gift nordic bug
Correct the gift send without settings for payments.

ecino commented on gift_compassion/models/sponsorship_gift.py:
```diff
@@ -228,6 +234,17 @@ def _compute_usd(self):
+        self.account_debit = param_obj.get_param("gift_expense_account_id")
+        self.journal_id = param_obj.get_param("gift_journal_id")
+        self.analytic = param_obj.get_param("gift_analytic_id")
+        self.analytic_tag = param_obj.get_param("gift_analytic_tag_id")
```
```suggestion
        for gift in self:
            gift.account_credit = param_obj.get_param("gift_income_account_id")
            gift.account_debit = param_obj.get_param("gift_expense_account_id")
            gift.journal_id = param_obj.get_param("gift_journal_id")
            gift.analytic = param_obj.get_param("gift_analytic_id")
            gift.analytic_tag = param_obj.get_param("gift_analytic_tag_id")
```

ecino commented on gift_compassion/models/sponsorship_gift.py:
```diff
@@ -228,6 +234,17 @@ def _compute_usd(self):
             gift.currency_usd = self.env.ref("base.USD")
 
+    def _compute_is_param_set(self):
+        self.is_param_set = all([int(self.account_credit), int(self.account_debit), int(self.journal_id)])
```
```suggestion
        for gift in self:
            gift.is_param_set = all([int(self.account_credit), int(self.account_debit), int(self.journal_id)])
```

Commit:
T0096 gift nordic date improvement
use the invoice_date instead of the date of the gift. So we can send the gift right away and don't have to wait.

simgonzalez committed:
Update gift_compassion/models/sponsorship_gift.py

Co-authored-by: ecino <ecino@users.noreply.github.com>

simgonzalez committed:
Update gift_compassion/models/sponsorship_gift.py

Co-authored-by: ecino <ecino@users.noreply.github.com>

Commit:
Different company can be possible
